src/main/java/pl/javastart/task/QueueManager.java

```diff
@@ -0,0 +1,131 @@
+package pl.javastart.task;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.Scanner;
+
+public class QueueManager {
+    private Queue<Vehicle> vehicleQueue = new LinkedList<>();
+    private Scanner sc = new Scanner(System.in);
+    private File queueFile = new File("src/main/resources/queue.txt");
+
+    void mainLoop() throws IOException {
+        Queue<Vehicle> vehicleYesterdayQueue;
+        vehicleYesterdayQueue = ReadWriteData.readFile(queueFile);
+        if (!vehicleYesterdayQueue.isEmpty()) {
+            vehicleQueue = vehicleYesterdayQueue;
+        }
+
+        Option option;
+        do {
+            printOptions();
+            System.out.println("Wybierz opcję: ");
+            option = Option.createFromInt(sc.nextInt());
+            sc.nextLine();
+            switch (option) {
+                case ADD:
+                    Vehicle vehicle;
+                    do {
+                        vehicle = readAndAddVehicle();
+                    } while (vehicle == null);
+
+                    vehicleQueue.offer(vehicle);
+                    System.out.println("Pojazd dodany do kolejki");
+                    break;
+                case TAKE:
+                    takeVehicle();
+                    break;
+                case EXIT:
+                    checkQueue();
+                    System.out.println("Koniec pracy na dziś.");
+                    break;
+            }
+
+        } while (option != Option.EXIT);
+    }
+
+    private void checkQueue() throws IOException {
+        if (!vehicleQueue.isEmpty()) {
```

Comment:
Zwróć uwagę na to, że jeżeli kolejka będzie pusta, to i tak warto nadpisać plik pustą zawartością. W innym przypadku będziemy mieli taką sytuację, że przy kolejnym uruchomieniu wczytamy z pliku pojazdy, które były już obsłużone w tym uruchomieniu programu